build-system/check-package-manager.js

```diff
@@ -154,6 +154,7 @@ function getYarnStableVersion(infoJson) {
 }
 
 function checkGlobalGulp() {
+  const firstInstall = getStdout('ls').trim().match(/node_modules/) === null;
```

Comment:
You can use the `fs` module to get the file list. Calling `ls` won't work on Windows :)

Reply:
Since this issue it's trying to fix a bug that happens before any of them are installed, turns out we can't import/use ANY node modules in this file. Trying to use a module breaks the script when it's the first install. Will use the `process.platform` strategy you guys used in `exec.js`